src/runtime/opencl.cpp

```diff
@@ -84,52 +84,116 @@ WEAK char device_type[256];
 WEAK int device_type_lock = 0;
 WEAK bool device_type_initialized = false;
 
+WEAK char build_options[1024];
+WEAK int build_options_lock = 0;
+WEAK bool build_options_initialized = false;
+
 }}}} // namespace Halide::Runtime::Internal::OpenCL
 
 using namespace Halide::Runtime::Internal::OpenCL;
 
 // Allow OpenCL 1.1 features to be used.
 #define ENABLE_OPENCL_11
 
-extern "C" {
-
-WEAK void halide_opencl_set_platform_name(const char *n) {
+namespace {
+void halide_opencl_set_platform_name_internal(const char *n) {
     if (n) {
-        strncpy(platform_name, n, 255);
+        size_t buffer_size = sizeof(platform_name) / sizeof(platform_name[0]);
+        strncpy(platform_name, n, buffer_size);
+        if (strlen(n) > buffer_size - 1) {
```

Comment:
nit: strncpy() actually zeros out the part of the buffer that it doesn't copy into, so I suspect it would be safe to just do `platform_name[buffer_size - 1] = 0` without the if